security/signup-controller.php

```diff
@@ -0,0 +1,39 @@
+<?php
+include "db-connection.php";
+
+$conn = openCon();
+
+$firstname = isset($_POST['firstname']) ? $_POST['firstname'] : false;
+$password = isset($_POST['password']) ? $_POST['password'] : false;
+$role = isset($_POST['role']) ? $_POST['role'] : false;
+
+if ($firstname && $password && $role)
+{
+  if ( strcmp($_POST['role'], "admin")==0 ) die("You are not authorized to do this.");
+  $sql="SELECT COUNT(*) FROM person WHERE email = :emailPlaceholder";
```

Comment:
Ok, now this is a correct way of using prepared statement. Good job, please apply on the other sql statements too.